Calculator/Calculator/Model/Formula.swift

```diff
@@ -0,0 +1,29 @@
+//
+//  Formula.swift
+//  Calculator
+//
+//  Created by Hemg on 2023/06/06.
+//
+
+struct Formula {
+    let operands: CalculatorItemQueue<Double>
+    let operators: CalculatorItemQueue<Operator>
+    
+    func result() -> Double {
+        let defaultValue: Double = 0.0
+        guard let firstOperand = operands.dequeueItem() else {
+            return defaultValue
+        }
+        
+        var result = firstOperand
+        
+        while !operators.isEmpty() {
+            guard let operatorValue = operators.dequeueItem() else { continue }
+            guard let nextOperand = operands.dequeueItem() else { continue }
+
+            result = operatorValue.calculate(result, nextOperand)
+        }
```

Comment:
while문을 통한 Optional Binding을 해볼 수 있겠네요.
[Optional Binding](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/thebasics/#Optional-Binding)을 보면 
> Optional binding can be used with if and while statements to check for a value inside an optional...

라고 명시되어 있습니다. 😀

```suggestion
        while let operatorValue = operators.dequeueItem(), let nextOperand = operands.dequeueItem() {
            result = operatorValue.calculate(result, nextOperand)
        }
```

Reply:
- 위의 코멘트에서와 같은 메서드 인데 밈이 알려주신 덕분에 코드가 더욱더 간결해졌습니다. 감사합니다!
`while let` 을 이렇게 써보게 되었습니다!! 감사합니다